server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -75,12 +79,15 @@
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
         Preconditions.checkArgument(request instanceof GetMailboxesRequest);
         GetMailboxesRequest mailboxesRequest = (GetMailboxesRequest) request;
-        return Stream.of(
+        TimeMetric timeMetric = metricFactory.timer(METHOD_NAME.getName());
+        Stream<JmapResponse> responses = Stream.of(
                 JmapResponse.builder().clientId(clientId)
                 .response(getMailboxesResponse(mailboxesRequest, mailboxSession))
                 .properties(mailboxesRequest.getProperties().map(this::ensureContainsId))
                 .responseName(RESPONSE_NAME)
                 .build());
+        timeMetric.elapseTimeInMs();
+        return responses;
```

Comment:
Couldn't you factorize all JMAP timeout process here:
```
    private Function<Method, Stream<JmapResponse>> extractAndProcess(AuthenticatedProtocolRequest request) {
        MailboxSession mailboxSession = request.getMailboxSession();
        return (Method method) -> {
                    try {
                        JmapRequest jmapRequest = jmapRequestParser.extractJmapRequest(request, method.requestType());
                        return method.process(jmapRequest, request.getClientId(), mailboxSession);
                    } catch (IOException e) {
                        LOGGER.error("Error occured while parsing the request.", e);
                        if (e.getCause() instanceof NotImplementedException) {
                            return errorNotImplemented(request);
                        }
                        return error(request, ErrorResponse.builder()
                                                        .type("invalidArguments")
                                                        .description(e.getMessage())
                                                        .build());
                    } catch (NotImplementedException e) {
                        return errorNotImplemented(request);
                    }
                };
    }
```
in RequestHandler class ?